### PR TITLE
feat(llm): add classify and pooling model support

### PR DESCRIPTION
**Pull request by jh-nv**
## Summary

- Add OpenAI-compatible classify and pooling request/response protocols and aggregators.
- Add classify/pooling model types, discovery, and Python binding support.
- Keep the feature dormant: this layer adds neither HTTP routes nor backend activation.
- This is PR 1 of 4 and consolidates drafts #12122–#12124.

## Validation

- This commit's pre-commit hooks passed.
- On the complete four-PR stack: 20 classify and 20 pooling Rust tests passed.
- The PyO3 binding built successfully with `maturin develop --uv`.
- The final tree exactly matches a clean merge of the former 14-PR stack onto current `main`.

## Stack

Merge in this order:

1. [#12140 — core protocols and model discovery](https://github.com/ai-dynamo/dynamo/pull/12140) **(this PR)**
2. [#12139 — frontend HTTP integration](https://github.com/ai-dynamo/dynamo/pull/12139)
3. [#12142 — vLLM worker integration and activation](https://github.com/ai-dynamo/dynamo/pull/12142)
4. [#12141 — examples and end-to-end validation](https://github.com/ai-dynamo/dynamo/pull/12141)

The original monolithic #12058 remains open; this four-PR stack replaces only the earlier 14-draft split (#12122–#12135).

## Related Issues

**🔗 This PR is linked to an issue:**

- Relates to DIS-2528

## Summary by CodeRabbit

* **New Features**
  * Added OpenAI-compatible classification and pooling endpoints, supporting text and token inputs.
  * Added classification results with labels, probabilities, and usage details.
  * Added pooling outputs in vector, matrix, and encoded formats.
  * Added model capability checks and discovery for classification, embedding, chat, and pooling support.
  * Added streaming response aggregation with structured error handling.
  * Added prompt truncation direction options for left- or right-side truncation.

### CODE REVIEW

File: lib/llm/src/model_type.rs
[the remark this answers is not in the document]
Comment (jh-nv) [reply lib/llm/src/model_type.rs:1]: Confirmed. This is the intentional PR 1 boundary: the core layer defines and discovers the model capabilities but does not activate HTTP surfaces. PR #12139 adds `EndpointType::Classify` / `EndpointType::Pooling` and maps both `ModelType` bits in `as_endpoint_types_with_anthropic`, so the complete stack enables the endpoints. Keeping the mapping out of this PR preserves the dormant-layer contract described in the PR body.

File: lib/bindings/python/src/dynamo/_core.pyi [hunk 1]
```diff
@@ -1649,6 +1649,11 @@ class ModelType:
     Audios: ModelType
     Videos: ModelType
     Realtime: ModelType
+    # Sequence-classification / cross-encoder pooling models served on /v1/classify.
+    Classify: ModelType
+    # Raw pooler output served on /v1/pooling (token embeddings, logits, rewards).
+    # Usually combined with Classify or Embedding: ModelType.Classify | ModelType.Pooling.
+    Pooling: ModelType
```
[the remark this answers is not in the document]
Comment (jh-nv) [reply lib/bindings/python/src/dynamo/_core.pyi:1661]: Fixed in 2ac446bad0086ce09517ed9e8f660b70383a2986. The class-level Python stub documentation now lists `Classify`, `Pooling`, `TensorBased`, and `Audios` along with the other supported surfaces.

File: lib/llm/src/discovery/watcher.rs [hunk 2]
```diff
@@ -1840,16 +1864,50 @@ impl ModelWatcher {
...
+                let push_router = PushRouter::<
+                    NvCreatePoolingRequest,
+                    Annotated<NvCreatePoolingResponse>,
+                >::from_client_with_monitor(
+                    client.clone(), router_config.router_mode, None
+                )
+                .await?;
+                worker_set.pooling_engine = Some(Arc::new(push_router));
+            }
         }
```
[the remark this answers is not in the document]
Comment (jh-nv) [reply lib/llm/src/discovery/watcher.rs]: Fixed in 2ac446bad0086ce09517ed9e8f660b70383a2986. Text-input surface construction is now additive: every advertised `ModelType` capability gets its own engine instead of selecting one mutually exclusive branch. This covers Chat+Pooling and the same masking risk for Completions and multimodal/realtime combinations. Added `text_pooling_family_preserves_chat_engine`, which registers `Chat | Classify | Pooling` and asserts all three engines; the watcher suite passes (19/19).

**Comment by jh-nv** [comment]
Review-body follow-up:

- Added `pooling::aggregator::tests::preserves_typed_stream_errors`, mirroring the classify coverage and preserving the structured `BackendError::InvalidArgument`.
- Kept `aggregate_stream` and `aggregate_typed_stream` separate. Delegating the legacy string helper to the typed helper and mapping the error back to `String` is not behavior-preserving for comment-only errors: it changes `message` to `Unknown: message`. A shared internal loop could preserve both contracts, but that refactor is unrelated to this feature and would add risk without changing functionality.

Validation on the updated stack: 19 watcher tests, 5 pooling aggregator tests, focused payload tests, `cargo clippy -p dynamo-llm --tests -- -D warnings`, Python syntax checks, and the classify launch script syntax check all pass.

**Review by rmccorm4 — APPROVED** [review]
